engine/src/main/java/org/pentaho/di/pan/PanCommandExecutor.java

```diff
@@ -326,10 +327,18 @@ public Trans loadTransFromRepository( Repository repository, String dirName, Str
     return trans; // return transformation loaded from the repo
   }
 
-  public Trans loadTransFromFilesystem( final String initialDir, final String filename, final String jarFilename ) throws Exception {
+  public Trans loadTransFromFilesystem( String initialDir, String filename, String jarFilename, String base64Zip )
+    throws Exception {
 
     Trans trans = null;
 
+    if ( !Utils.isEmpty( base64Zip ) ) {
+      //expected form of filename "*.ktr"
+      String zipPath = Const.getUserHomeDirectory() + File.separator + java.util.UUID.randomUUID().toString() + ".zip";
+      filename = "zip:file:" + File.separator + zipPath + "!" + filename;
```

Comment:
Shouldn't this be 

```
"zip:file:" + File.separator + File.separator + zipPath + ... 
```

?

I see only 1 `File.separator` ...

Reply:
yes, somehow it was working, I dunno, but I will change it